packages/vercel/src/lib/redirects.ts

```diff
@@ -45,25 +45,28 @@ function getParts(part: string, file: string) {
 // 2022-04-26
 function getMatchPattern(segments: RoutePart[][]) {
 	return segments
-		.map((segment) => {
-			return segment[0].spread
+		.map((segment, segmentIndex) => {
+			return segment.length === 1 && segment[0].spread
 				? '(?:\\/(.*?))?'
-				: segment
-						.map((part) => {
-							if (part)
-								return part.dynamic
-									? '([^/]+?)'
-									: part.content
-											.normalize()
-											.replace(/\?/g, '%3F')
-											.replace(/#/g, '%23')
-											.replace(/%5B/g, '[')
-											.replace(/%5D/g, ']')
-											.replace(/[*+?^${}()|[\]\\]/g, '\\$&');
-						})
-						.join('');
+				: (segmentIndex === 0 ? '' : '/') +
```

Comment:
I noticed this check differs from the core implementation. I believe you said this address the case where a spread param lacks a leading slash (e.g. `/path[...spread]`). Is there a reason Vercel needs special handling?

Reply:
@bholmesdev great question!
Not exactly. This a replacement for `join('/')` defined here: https://github.com/withastro/adapters/blob/2246e6e263ba5d4c67912894c011dc64cd906634/packages/vercel/src/lib/redirects.ts#L66
It means "join all route segments with `'/'`, but omit it if the segment is a spread" (e.g. "/path/**[...spread]**").

And the case you are describing is handled here (I should probably cover it with tests as well): https://github.com/withastro/adapters/blob/2625aa3d049b3c883b07017e29bace4baff979f5/packages/vercel/src/lib/redirects.ts#L56

Initially I copied the core implementation of `getPattern()` method and tried to use that, but I realized that it would require more code changes around redirects. I've tried to achieve the goal without changing the interface of `getMatchPattern()` function. If you think that the correct way is to copy `getPattern()` from core and adjust the rest of the Vercel adapter code accordingly, let me know!

Reply:
Got it. I can see why we took this route to avoid a larger refactor. I think that's the right move for this PR! I'd consider exposing a utility for this from `astro` core to consolidate code in the future.